Content.Server/Traitor/Uplink/UplinkSystem.cs

```diff
@@ -1,82 +1,130 @@
-        /// <param name="uplinkEntity">The entity that will actually have the uplink functionality. Defaults to the PDA if null.</param>
-        /// <returns>Whether or not the uplink was added successfully</returns>
-        public bool AddUplink(EntityUid user, FixedPoint2? balance, EntityUid? uplinkEntity = null)
+        // Try to find target item
+        if (uplinkEntity == null)
         {
-            // Try to find target item
+            uplinkEntity = FindUplinkTarget(user);
             if (uplinkEntity == null)
-            {
-                uplinkEntity = FindUplinkTarget(user);
-                if (uplinkEntity == null)
-                    return false;
-            }
+                return ImplantUplink(user, balance);
+        }
 
-            EnsureComp<UplinkComponent>(uplinkEntity.Value);
-            var store = EnsureComp<StoreComponent>(uplinkEntity.Value);
-            store.AccountOwner = user;
-            store.Balance.Clear();
-            if (balance != null)
-            {
-                store.Balance.Clear();
-                _store.TryAddCurrency(new Dictionary<string, FixedPoint2> { { TelecrystalCurrencyPrototype, balance.Value } }, uplinkEntity.Value, store);
-            }
+        EnsureComp<UplinkComponent>(uplinkEntity.Value);
+
+        SetUplink(user, uplinkEntity.Value, balance);
 
-            // TODO add BUI. Currently can't be done outside of yaml -_-
+        // TODO add BUI. Currently can't be done outside of yaml -_-
+
+        return true;
+    }
 
-            return true;
+    /// <summary>
+    /// Configure TC for the uplink
+    /// </summary>
+    public void SetUplink(EntityUid user, EntityUid uplink, FixedPoint2? balance)
+    {
+        var store = EnsureComp<StoreComponent>(uplink);
+        store.AccountOwner = user;
+        store.Balance.Clear();
+        if (balance != null)
+        {
+            store.Balance.Clear();
+            _store.TryAddCurrency(new Dictionary<string, FixedPoint2> { { TelecrystalCurrencyPrototype, balance.Value } }, uplink, store);
         }
+    }
 
-        /// <summary>
-        /// Finds the entity that can hold an uplink for a user.
-        /// Usually this is a pda in their pda slot, but can also be in their hands. (but not pockets or inside bag, etc.)
-        /// </summary>
-        public EntityUid? FindUplinkTarget(EntityUid user)
+    /// <summary>
+    /// Implant an uplink as a fallback measure if the traitor had no PDA
+    /// </summary>
+    public bool ImplantUplink(EntityUid user, FixedPoint2? balance)
+    {
+        var implants = new List<string>(){FallbackUplinkImplant};
+
+        if (!_proto.TryIndex<ListingPrototype>(FallbackUplinkCatalog, out var catalog))
+            return false;
+
+        if (!catalog.Cost.TryGetValue(TelecrystalCurrencyPrototype, out var cost))
+            return false;
+
+        var implantCost = cost.Int();
+
+        _subdermalImplant.AddImplants(user, implants);
```

Comment:
In my solution instead of having to search through the containers for the implant, I just made an AddImplant function for a single implant and had it return the implant entity
![grafik](https://github.com/user-attachments/assets/442fec8c-4568-4873-ba9a-ec407d60e6b4)

Reply:
I still think returning the entities from the AddImplants function is cleaner than looking through all implants to search for them like you do below. And other systems adding implants might be able to use that as well.

Reply:
Ok you sold me, but is there a reason there is no null return if the implant was invalid?

Reply:
Good catch!